### ISSUE TITLE
On latest Vivaldi SIngleFile forgets last used folder.

### ISSUE THREAD

**Issue by janaxhell** [gildas-lormeau/SingleFile#1406]
**Describe the bug**
Before a couple of Vivaldi builds ago, SingleFIle used to remember last used folder, now it doesn't. I keep being offered the default system Download folder. Vivaldi is set as **Update Default Location when choosing Save As location**, which translates to "remember last used folder". I obviously haven't touched that setting, something in the latest builds must have broken SingleFile extension.

**To Reproduce**
1. Save a page to a specific folder.
2. Save another page to a specific folder.
3. When dialog opens you are not offered the previous folder, but the default system folder.

**Expected behavior**
I should be offered the same folder I used when saving the previous page.

**Environment**
 - OS: Win10 Pro
 - Browser: Vivaldi
 - Version: 6.6.3271.48 (Stable channel) (64-bit)

**Comment by gildas-lormeau** [comment]
Can you please try to disable `Misc. > save pages in background` in the options page of SingleFile?

**Comment by janaxhell** [comment]
Yes, changing that makes SingleFile remember last used folder, but now every time I save a page this (useless) dialog appears:

![image](https://github.com/gildas-lormeau/SingleFile/assets/55061479/05180bfe-11d3-4eab-b221-d8125f266099)

**Comment by gildas-lormeau** [comment]
Did you enable `File name > open the "Save as" dialog to confirm the file name`? If this is the case, please disable it.

**Comment by janaxhell** [comment]
Ahh, yes, now it works fine again. But I didn't touch SingleFile options at all before this issue. Could it possibly crash at some point and reset itself? Sometimes I see notifications stating "Extension X has crashed, please click to reload". Maybe it happened when I wasn't watching.

**Comment by gildas-lormeau** [comment]
A crash should not reset the options. I'll try to do some tests to see if Vivaldi changed something that would explain the issue.
